### PR TITLE
update to support no iso-8859-15 language

### PR DESCRIPTION
**Pull request by knorth55**
update https://github.com/ros-drivers/audio_common/pull/154 to support for non iso-8859-15 language
I checked with our japanese speaking robots.

- python2
 - [x] iso-8859-15 language
 - [x] non iso-8859-15 language
- python3
 - [x] iso-8859-15 language
 - [x] non iso-8859-15 language

### CODE REVIEW

File: sound_play/scripts/soundplay_node.py [hunk 1]
```diff
@@ -254,11 +254,15 @@ def select_sound(self, data):
...
+                        if hasattr(data.arg, 'decode'):
+                            txtfile.write(data.arg.decode('UTF-8').encode('ISO-8859-15'))
+                        else:
+                            txtfile.write(data.arg.encode('ISO-8859-15'))
                     except UnicodeEncodeError:
-                        txtfile.write(data.arg)
-                    except AttributeError:
-                        txtfile.write(data.arg.encode('ISO-8859-15'))
+                        if hasattr(data.arg, 'decode'):
+                            txtfile.write(data.arg.decode('UTF-8').encode('UTF-8'))
```
Comment (mikaelarguedas) [inline sound_play/scripts/soundplay_node.py]: Does this line have any effect ?
Maybe we could just:
```suggestion
                            txtfile.write(data.arg)
```
in that case?

**Comment by mikaelarguedas** [comment]
This branch has the same problem for me for Python3 non iso-8859-15 language:

```
Segmentation fault (core dumped)
(/sound_play) [ERROR] [1599216737.437369]: Sound synthesis failed. Is festival installed? Is a festival voice installed? Try running "rosdep satisfy sound_play|sh". Refer to http://wiki.ros.org/sound_play/Troubleshooting
(/sound_play) [ERROR] [1599216737.438651]: Exception in callback: 'NoneType' object has no attribute 'command'
(/sound_play) [INFO] [1599216737.439686]: Traceback (most recent call last):
  File "/tmp/ws/src/audio_common/sound_play/scripts/soundplay_node.py", line 313, in callback
    sound.command(data.command)
AttributeError: 'NoneType' object has no attribute 'command'
```

If it works for you that's ok with me as my use cases will not enter this specific code path

File: sound_play/scripts/soundplay_node.py (hunk 1 above)
Comment (knorth55) [reply sound_play/scripts/soundplay_node.py]: yes, you are right.
thank you, I updated.

**Comment by knorth55** [comment]
```
Segmentation fault (core dumped)
(/sound_play) [ERROR] [1599216737.437369]: Sound synthesis failed. Is festival installed? Is a festival voice installed? Try running "rosdep satisfy sound_play|sh". Refer to http://wiki.ros.org/sound_play/Troubleshooting
(/sound_play) [ERROR] [1599216737.438651]: Exception in callback: 'NoneType' object has no attribute 'command'
(/sound_play) [INFO] [1599216737.439686]: Traceback (most recent call last):
  File "/tmp/ws/src/audio_common/sound_play/scripts/soundplay_node.py", line 313, in callback
    sound.command(data.command)
AttributeError: 'NoneType' object has no attribute 'command'
```

This error is because festival does not support japanese.
We use `voice_text` to overwrite `text2wave` (which is festival command) like this.
https://github.com/jsk-ros-pkg/jsk_3rdparty/blob/master/3rdparty/voice_text/bin/text2wave
and set this `text2wave` in `PATH` like this.
https://github.com/jsk-ros-pkg/jsk_3rdparty/blob/master/3rdparty/voice_text/launch/voice_text.launch#L24-L30
We also need paid license for this `voice_text` so it is really tricky to speak japanese.
But I tested this PR with our robot with paid license, and he speaks japanese correctly.

**Review by mikaelarguedas — APPROVED** [review]
Sounds good to me :+1:, thanks for the additionnal information

**Pull request merged**